Document PR exception tracking process. (#6947)
sourcegraph/handbook: content/departments/engineering/dev/process/pullrequest-compliance.md

# Pull Requests compliance and requirements

All pull requests must provide [a test plan](https://docs.sourcegraph.com/dev/background-information/testing_principles#test-plans) and be reviewed by a teammate. If for situational reasons, it needs to be exempted, an [exception is automatically recorded](https://docs.sourcegraph.com/dev/background-information/testing_principles#exceptions).

Exceptions are periodically reviewed by the @dev-experience-team on a weekly basis.

## Contact

Tag @dev-experience-team on Slack for help.

Discuss the process in #ask-dev-experience.

## Process

### For all teammates

> 1. If a PR is considered to be an exception, a GitHub issue is automatically created over [a GitHub repository dedicated to track them](https://github.com/sourcegraph/sec-pr-audit-trail/issues), and the teammate who caused it is assigned to it.

Reviewer: ```suggestion
1. If a PR is considered to be an exception, a GitHub issue is automatically created over [a GitHub repository dedicated to track them](https://github.com/sourcegraph/sec-pr-audit-trail/issues), and the teammate who is responsible will be assigned.
```

> 1. When assigned such an issue, the teammate (or a peer) has to post an explanation for the exception as well as to justify how it was reviewed and/or tested, and close it.

Reviewer: ```suggestion
1. When assigned such an issue, the teammate (or a peer) is required to post an explanation for the exception as well as to justify how it was reviewed and/or tested, and close it.
```

1. If the issue is reopened by @dev-experience-support, the teammate must update the explanation until considered valid by @dev-experience-support.

> ### For engineer on dev-experience-support roation

Reviewer 2: ```suggestion
### For engineer on dev-experience-support rotation
```
